apps/gitness/src/pages-v2/pull-request/pull-request-compare.tsx

```diff
@@ -0,0 +1,382 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useNavigate, useParams } from 'react-router-dom'
+
+import * as Diff2Html from 'diff2html'
+import { useAtom } from 'jotai'
+import { compact, isEqual } from 'lodash-es'
+
+import {
+  CreateRepositoryErrorResponse,
+  mergeCheck,
+  OpenapiCreatePullReqRequest,
+  TypesBranchExtended,
+  TypesCommit,
+  useCreatePullReqMutation,
+  useDiffStatsQuery,
+  useFindRepositoryQuery,
+  useGetPullReqByBranchesQuery,
+  useListBranchesQuery,
+  useListCommitsQuery,
+  useListTagsQuery,
+  useRawDiffQuery
+} from '@harnessio/code-service-client'
+import { SkeletonList } from '@harnessio/ui/components'
+import { BranchSelectorListItem, BranchSelectorTab, CompareFormFields, PullRequestCompare } from '@harnessio/ui/views'
+
+import { useGetRepoRef } from '../../framework/hooks/useGetRepoPath'
+import { useTranslationStore } from '../../i18n/stores/i18n-store'
+import { parseSpecificDiff } from '../../pages/pull-request/diff-utils'
+import { changesInfoAtom, DiffFileEntry, DiffViewerExchangeState } from '../../pages/pull-request/types/types'
+import { changedFileId, DIFF2HTML_CONFIG, normalizeGitFilePath } from '../../pages/pull-request/utils'
+import { PathParams } from '../../RouteDefinitions'
+import { normalizeGitRef } from '../../utils/git-utils'
+
+/**
+ * TODO: This code was migrated from V2 and needs to be refactored.
+ */
+export const CreatePullRequest = () => {
+  const createPullRequestMutation = useCreatePullReqMutation({})
+  const { repoId, spaceId, diffRefs } = useParams<PathParams>()
+  const [isBranchSelected, setIsBranchSelected] = useState<boolean>(diffRefs ? true : false) // State to track branch selection
+
+  const [diffTargetBranch, diffSourceBranch] = diffRefs ? diffRefs.split('...') : [undefined, undefined]
+
+  const navigate = useNavigate()
+  const [apiError, setApiError] = useState<string | null>(null)
+  const repoRef = useGetRepoRef()
+  const [selectedTargetBranch, setSelectedTargetBranch] = useState<BranchSelectorListItem>(
+    diffTargetBranch ? { name: diffTargetBranch, sha: '' } : { name: 'main', sha: '' }
+  )
+  const [selectedSourceBranch, setSelectedSourceBranch] = useState<BranchSelectorListItem>(
+    diffSourceBranch ? { name: diffSourceBranch, sha: '' } : { name: 'main', sha: '' }
+  )
+  const [prBranchCombinationExists, setPrBranchCombinationExists] = useState<number | null>(null)
+  const commitSHA = '' // TODO: when you implement commit filter will need commitSHA
+  const defaultCommitRange = compact(commitSHA?.split(/~1\.\.\.|\.\.\./g))
+  const [
+    commitRange
+    //  setCommitRange  TODO: add commit view filter dropdown to manage different commits
+  ] = useState(defaultCommitRange)
+  const targetRef = useMemo(() => selectedTargetBranch.name, [selectedTargetBranch])
+  const sourceRef = useMemo(() => selectedSourceBranch.name, [selectedSourceBranch])
+  const [cachedDiff, setCachedDiff] = useAtom(changesInfoAtom)
```

Comment:
Why we are using `jotai` here? Is this expected? We have `zustand` being used already, can we adopt that here and remove `jotai` from entire application?

Reply:
we wont use jotai in the future i was just migrating the compare page over. Will remove that in next pr. Just wanted to break up pr before it got too large @praneshg239